lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -91,29 +91,75 @@ end)
 /******************************************************************************/
 -- Name functions
 
--- Set the name of the E2 itself
-e2function void setName( string name )
-	local e = self.entity
-	if (e.name == name) then return end
-	if (name == "generic" or name == "") then
+local CurTime = CurTime
+
+local antiSpamLookup = {}
+local function NameSpamCheck(ent) -- Returns true if it is being spammed; otherwise false
+	if antiSpamLookup[ent] and antiSpamLookup[ent] > CurTime() then
+		return true
+	end
+	antiSpamLookup[ent] = CurTime() + 0.1 -- Adjust the delay here, it is applied to setter (setName/setComponentName) functions
+	return false
+end
+
+local function SetE2Name(ent, name)
+	if NameSpamCheck(ent) or ent.name == name then return end
+	if name == "generic" or name == "" then -- NOTE: This could be simplified if we exclude the "generic" check
 		name = "generic"
-		e.WireDebugName = "Expression 2"
+		ent.WireDebugName = "Expression 2"
 	else
-		e.WireDebugName = "E2 - " .. name
+		ent.WireDebugName = "E2 - " .. name
+	end
+	ent.name = name
+	ent:SetNWString("name", name)
+	ent:SetOverlayText(name)
+end
+
+__e2setcost(20)
+
+local IsValid = IsValid
+
+-- Set the name of the E2 itself
+e2function void setName(string name)
+	local e = self.entity
+	if IsValid(e) then
+		SetE2Name(e, name)
+	end
+end
+
+-- Set the name of another E2
+e2function void entity:setName(string name)
```

Comment:
> If you are worried about data signals, you might be using them wrong

That's the thing, I don't even want to do data-signal just so I could trigger the `setName` call in another E2.. With this function, we will be able to set the name of another E2 straightaway _at no cost_. No point in arguing about it; I doubt you will convince me to remove it.

-----

Update incoming.